lib/hw1/Docker homework.ipynb

```diff
@@ -93,22 +93,226 @@
+      "\n",
+      "3 packages are looking for funding\n",
+      "  run `npm fund` for details\n",
+      "\n",
+      "found 0 vulnerabilities\n",
+      "\u001b[91mnpm\u001b[0m\u001b[91m \u001b[0m\u001b[91mnotice\u001b[0m\u001b[91m \n",
+      "\u001b[0m\u001b[91mnpm\u001b[0m\u001b[91m \u001b[0m\u001b[91mnotice\u001b[0m\u001b[91m New minor version of npm available! 7.13.0 -> 7.14.0\n",
+      "\u001b[0m\u001b[91mnpm\u001b[0m\u001b[91m \u001b[0m\u001b[91mnotice\u001b[0m\u001b[91m Changelog: <https://github.com/npm/cli/releases/tag/v7.14.0>\n",
+      "\u001b[0m\u001b[91mnpm\u001b[0m\u001b[91m \u001b[0m\u001b[91mnotice\u001b[0m\u001b[91m Run `npm install -g npm@7.14.0` to update!\n",
+      "\u001b[0m\u001b[91mnpm\u001b[0m\u001b[91m \u001b[0m\u001b[91mnotice\u001b[0m\u001b[91m \n",
+      "\u001b[0mRemoving intermediate container e27ad36880ad\n",
+      " ---> f06ec0bcbfd9\n",
+      "Step 6/7 : CMD [\"cd\", \"/usr/basic_server\"]\n",
+      " ---> Running in ebffb3f5c4a2\n",
+      "Removing intermediate container ebffb3f5c4a2\n",
+      " ---> 625daa9ac78a\n",
+      "Step 7/7 : CMD [\"node\", \"index.js\"]\n",
+      " ---> Running in 525005de9a6e\n",
+      "Removing intermediate container 525005de9a6e\n",
+      " ---> 1528971503eb\n",
+      "Successfully built 1528971503eb\n",
+      "Successfully tagged basic-server:latest\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Add the bash commands for task 1 here\n",
+    "cd /usr/project/basic-server\n",
+    "docker build -t basic-server ."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sending build context to Docker daemon  3.584kB\n",
+      "Step 1/2 : FROM mariadb:latest\n",
+      "latest: Pulling from library/mariadb\n",
+      "\n",
+      "\u001b[1B3491a907: Pulling fs layer \n",
+      "\u001b[1B1312ef6f: Pulling fs layer \n",
+      "\u001b[1B50ddb7d0: Pulling fs layer \n",
+      "\u001b[1B2e2ff778: Pulling fs layer \n",
+      "\u001b[1Ba7dc2af9: Pulling fs layer \n",
+      "\u001b[1Bf4f4774a: Pulling fs layer \n",
+      "\u001b[1B9f940bd2: Pulling fs layer \n",
+      "\u001b[1B6ea8f4b9: Pulling fs layer \n",
+      "\u001b[1Bc4d334a0: Pulling fs layer \n",
+      "\u001b[1Bf65594e2: Pulling fs layer \n",
+      "\u001b[1Bd5c12f84: Pulling fs layer \n",
+      "\u001b[1BDigest: sha256:018ee5055ef34fd95dae490266da232cd527625704c6badf84006b36f49ad265\n",
+      "Status: Downloaded newer image for mariadb:latest\n",
+      " ---> 2a2c18b8e036\n",
+      "Step 2/2 : COPY setup.sql /docker-entrypoint-initdb.d/setup.sql\n",
+      " ---> 650192f03be4\n",
+      "Successfully built 650192f03be4\n",
+      "Successfully tagged mymariadb:latest\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Add the bash commands for task 1 here\n",
+    "cd /usr/project/mariadb\n",
+    "docker build -t mymariadb ."
```

Comment:
Could have also used a mount to accomplish this.

As in mounting the basic-server as the `docker-entrypoint-initdb.d`

Reply:
I know but the mounting operation failed in the jupyter container. (outside it was working fine)

Reply:
This is probably because docker runs on the host file system, not the containers.